src/js/otus/hooks.ts

```diff
@@ -0,0 +1,38 @@
+import { useFetchOTU } from "@otus/queries";
+import { OTUIsolate } from "@otus/types";
+import { indexOf, map, sortBy } from "lodash-es/lodash";
+
+type SequencesResponse = {
+    data: OTUIsolate;
+    isLoading: boolean;
+};
+
+/**
+ * A hook for sorting the sequences for the active isolate
+ *
+ * @param otuId - The otu which the isolate belongs to
+ * @param isolate - The active isolate
+ */
+export default function useGetSequences(otuId: string, isolate: OTUIsolate): SequencesResponse {
+    const { data, isLoading } = useFetchOTU(otuId);
+
+    if (isLoading) {
+        return { data: data, isLoading: isLoading };
+    }
+
+    const { sequences } = isolate;
+
+    if (sequences) {
+        const segmentNames = map(data.schema, "name");
+        const updatedSequences = sortBy(sequences, [
+            entry => {
+                const index = indexOf(segmentNames, entry.segment);
+                return index !== -1 ? index : segmentNames.length;
+            },
+        ]);
+
+        return { data: { ...isolate, sequences: updatedSequences }, isLoading: isLoading };
+    }
+
+    return { data: { ...isolate, sequences: [] }, isLoading: isLoading };
+}
```

Comment:
I think we need a rethink on how this is being done. While this version is better since it no longer has the implicit dependency, I think as a long term design pattern having to include extra loaders is sub-optimal. 

Taking a look at the code, I think this could be good use case for using context. This way we can avoid the excess loaders and keep any dependencies explicit.

I think the template show here is a good jumping off point for getting this done nicely: https://tkdodo.eu/blog/react-query-and-react-context

Then for the actual sorting we can just use a helper called something along the lines of `sortSequnecesBySegment(sequences, segments): sequences`

Happy to talk over a call if you want additional guidance on this one